shims/spark311db/src/main/scala/com/nvidia/spark/rapids/shims/spark311db/GpuRunningWindowExec.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims.spark311db
+
+import com.databricks.sql.execution.window.RunningWindowFunctionExec
+import com.nvidia.spark.rapids.{BaseExprMeta, DataFromReplacementRule, GpuExec, GpuOverrides, GpuWindowExec, RapidsConf, RapidsMeta, SparkPlanMeta}
+
+import org.apache.spark.sql.catalyst.expressions.{Expression, NamedExpression, SortOrder}
+
+/**
+ * GPU-based window-exec implementation, analogous to RunningWindowFunctionExec.
+ */
+class GpuRunningWindowExecMeta(runningWindowFunctionExec: RunningWindowFunctionExec,
```

Comment:
nit: Would it be possible to add in an abstract class that is the base for both `GpuRunningWindowExecMeta` and `GpuWindowExecMeta`? The main reason I ask is because there is a bit of code that is copied and pasted here, not all of it, but for my PR there is a lot more of it, and I would like to avoid duplicating code as much as possible.

Reply:
Ok, this is proving non-trivial right now. `RunningWindowFunctionExec` and `WindowExec` aren't related classes. `GpuBaseWindowExecMeta` will need to be genericized. And Scala's generics don't seem to work similarly to C++'s. I'm trying to sort this out now.

Reply:
I think I have figured out a way to do this. Testing now.

Reply:
Sorry for the delay. This warranted testing on Apache Spark, Databricks 7.3, and Databricks 8.2. This should be resolved now.